apps/fabric-website/src/SiteDefinition/SiteDefinition.pages/Controls/web.tsx

```diff
@@ -1,580 +1,340 @@
+    if (!foundCategory && pagesByCategory.Other) {
+      pagesByCategory.Other.push(_generatePage(requireContext, componentName, pagePath, {}));
+    }
+  }
+
+  // Handle the sub-pages
+  for (const parentComponentName of Object.keys(subPageCategories)) {
+    const parentCategory = subPageCategories[parentComponentName];
+    const subPages = parentCategory.subPages!;
+    // clean up the parent page: remove the subPages prop, which isn't used in the real nav
+    delete parentCategory.subPages;
+
+    const childPagePaths = componentSubPages[parentComponentName];
+    if (!childPagePaths.length) {
+      continue; // no pages in this category
+    }
+
+    const parentPage = parentCategory as INavPage;
+    // First page in the category is a non-category version of the parent page
+    parentPage.pages = [{ ...parentPage }];
+    parentPage.isCategory = true;
+
+    for (const pagePath of childPagePaths) {
+      // pagePath will be like ./DetailsList/DetailsListBasicPage
+      const pathParts = pagePath.split('/');
+      // DetailsListBasic
+      const pageComponentName = pathParts[2].replace(/Page$/, '');
+      // Basic
+      const pageName = pageComponentName.replace(parentComponentName, '');
+
+      const subPageOverrides = subPages[pageName];
+      if (subPageOverrides) {
+        parentPage.pages.push(
+          _generatePage(requireContext, pageComponentName, pagePath, {
+            ...subPageOverrides,
+            title: `${parentComponentName} - ${subPageOverrides.title || pageName}`,
+            url: `${parentComponentName}/${subPageOverrides.url || pageName}`.toLowerCase()
+          })
+        );
+      }
+    }
+  }
+
+  // Convert the categories to an array (filter out empty categories)
+  return categoryNames
+    .filter(category => !!pagesByCategory[category].length)
+    .map(category => {
+      const pages = pagesByCategory[category];
+      return {
+        title: category,
+        url: pages[0].url,
+        isCategory: true,
+        pages
+      };
+    });
+}
+
+function _generatePage(
+  requireContext: __WebpackModuleApi.RequireContext,
+  componentName: string,
+  pagePath: string,
+  overrides: CategoryPage
+): CategoryPage & INavPage {
+  const { url, ...nonUrlOverrides } = overrides;
+  return {
+    title: componentName,
+    url: '#/controls/web/' + (url || componentName.toLowerCase()),
+    component: () => <LoadingComponent title={overrides.title || componentName} />,
+    getComponent: cb => requireContext(pagePath).then((mod: any) => cb(mod[componentName + 'Page'])),
+    ...nonUrlOverrides
+  };
+}
+
+export const controlsPagesWeb: INavPage[] = [
+  {
+    title: 'Controls',
+    url: '#/controls/web',
+    isHiddenFromMainNav: true,
+    component: () => <LoadingComponent title="Controls" />,
+    getComponent: cb => require.ensure([], require => cb(require<any>('../../../pages/Overviews/ControlsPage/ControlsPage').ControlsPage))
+  },
+  ...generateCategories()
 ];
+
+// Old list
```

Comment:
Is this useful to keep around in the source? Looks like the same has been captured in `categories`?

Reply:
Not exactly. The categories in the new list are different, and one of the things I want to discuss later is whether the new list of categories is actually the "right" one.